File: backend/app/models/climate.py
from pydantic import BaseModel, Field


class Location(BaseModel):
    region: str
    lat: float | None = None
    lon: float | None = None


class ClimateRecordResponse(BaseModel):
    id: str
    source_type: str
    location: Location
    timestamp: str
    temperature_c: float | None = None
    precipitation_mm: float | None = None
    humidity_pct: float | None = None
    co2_ppm: float | None = None
    is_anomaly: bool = False
    is_archived: bool = False
    ingestion_id: str | None = None
    created_at: str


class PaginatedClimateRecords(BaseModel):
    items: list[ClimateRecordResponse]
    total: int
    page: int
    limit: int


class ClimateSummaryItem(BaseModel):
    region: str
    period: str
    avg_temperature_c: float | None = None
    total_precipitation_mm: float | None = None
    record_count: int
    anomaly_count: int


class ClimateSummaryResponse(BaseModel):
    items: list[ClimateSummaryItem]

File: frontend/src/types/climate.ts
export interface Location {
  region: string;
  lat: number | null;
  lon: number | null;
}

export interface ClimateRecord {
  id: string;
  source_type: string;
  location: Location;
  timestamp: string;
  temperature_c: number | null;
  precipitation_mm: number | null;
  humidity_pct: number | null;
  co2_ppm: number | null;
  is_anomaly: boolean;
  is_archived: boolean;
  ingestion_id: string | null;
  created_at: string;
}

export interface PaginatedClimateRecords {
  items: ClimateRecord[];
  total: number;
  page: number;
  limit: number;
}

export interface ClimateSummaryItem {
  region: string;
  period: string;
  avg_temperature_c: number | null;
  total_precipitation_mm: number | null;
  record_count: number;
  anomaly_count: number;
}

export interface ClimateSummaryResponse {
  items: ClimateSummaryItem[];
}